macros/src/bitflags.rs

```diff
@@ -0,0 +1,153 @@
+use proc_macro::TokenStream;
+use proc_macro2::TokenStream as TokenStream2;
+use quote::{quote, ToTokens};
+use syn::{
+    parse::{Parse, ParseStream},
+    parse_macro_input,
+    punctuated::Punctuated,
+    token, Attribute, Expr, Ident, Token, Type, Visibility,
+};
+
+use crate::{mksym, symbol::Symbol};
+
+pub(super) fn expand(ts: TokenStream) -> TokenStream {
+    let ts2 = proc_macro2::TokenStream::from(ts.clone());
+    let input: Input = parse_macro_input!(ts);
+
+    let codegen = Codegen::new(&input);
+
+    // Encode package and disambiguator to provide the decoder with all info it needs (even if
+    // technically redundant, since it's also stored in the symbol we create).
+    let format_str = format!(
+        "{{={}:__internal_bitflags_{}@{}@{}}}",
+        input.ty.to_token_stream(),
+        input.ident,
+        crate::symbol::package(),
+        crate::symbol::disambiguator(),
+    );
+    let sym = mksym(&format_str, "bitflags", false);
+
+    let ident = &input.ident;
+    let ty = input.ty;
+    let flags = codegen.flags.iter().map(|f| &f.def);
+    let result = quote! {
+        const _: () = {
+            fn assert<T: defmt::export::UnsignedInt>() {}
```

Comment:
doesn't upstream `bitflags!` already implement this check? or are we limiting what upstream `bitflags!` can do by only accepting unsigned ints here whereas upstream also accepts signed ints? (seems unlikely)

Reply:
The bitflags crate does not check this type, its docs just note

> The flags should only be defined for integer types, otherwise unexpected type errors may occur at compile time.

Reply:
then this is a great addition!